### PR TITLE
Scala3/kernel+core

### PR DESCRIPTION
**Pull request by Grryum**
Scala 3 is coming!

Here are modules already ported:
- kernel-higherKind (all except macros) - based on #1134 
- kernel (all RepresentableK instances derived by hand without hk macros)
- coreCE2
- coreCE3
- zio2Core

### CODE REVIEW

File: modules/kernel/src/main/scala-3/tofu/internal/Interop.scala [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            withExplicitArgs.appliedToArgs(summonedInst)
+          case _             =>
+            withExplicitArgs
+
+        // for some reason lambda type like this `FinallyCarrier2.Aux[F, Throwable, [x] =>> CE2Kernel.CEExit[java.lang.Throwable, x]]`
+        // has no subtyping relations with type `FinallyCarrier2.Aux[F, Throwable, [x] =>> Any]` and `.asExprOf[X]` throws java.lang.Exception: Expr cast exception
+
+        // For such cases old and classic `asInstanceOf` helps with extra casting.
+
+        val asInstanceOf = TypeRepr.of[AnyRef].typeSymbol.methodMember("asInstanceOf").head
```
Comment (Odomontois) [inline modules/kernel/src/main/scala-3/tofu/internal/Interop.scala:75]: That's very unpleasant. But I suppose we can cope with it for now

**Comment by Odomontois** [comment]
Thank you very much for all of this colossal work.
The only thing that frightens me is the proposed implementation of [`Interop` macros](https://github.com/tofu-tf/tofu/pull/1192/files#diff-048ed709a654c7dd5bf69a659b0e83a13168c83b480ca5090b179a6ca5431e12).
It clearly was done to be compatible with scala 2 definitions.
When I've implemented this I've missed some features of scala 3 inlines a lot, meaning I had to pass code line to the macro via structural type.
So I've had in mind that when that would be migrating to scala 3, we will rewrite the delegation implementation and "delegating" to instances would have scala-specificб but because of all that huge work, I hope we can do that in the folliwing PR.
So I have no options but to accept this PR, it feels like it's now or never

**Review by Odomontois — APPROVED** [review]

File: modules/kernel/src/main/scala-3/tofu/internal/Interop.scala (hunk 1 above)
Comment (Grryum) [reply modules/kernel/src/main/scala-3/tofu/internal/Interop.scala]: yup! @road21 tried to fix it and replaced with `widen`.
Hope now it is not so unpleasant :)

**Comment by Grryum** [comment]
partially closes #1068

**Comment by Odomontois** [comment]
What happened with doc generation?

**Pull request merged**